src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react';
+import { User } from '../../types/User';
+import { TodoUser } from '../../types/TodoUser';
+
+interface Props {
+  users: User[];
+  todos: TodoUser[];
+  onAdd: (newTodo: TodoUser[]) => void;
+}
+
+export const NewTodo: React.FC<Props> = ({ users, todos, onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [select, setSelect] = useState(0);
+
+  const [touched, setTouched] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasSelectError, setHasSelectError] = useState(false);
+
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  const reset = () => {
+    setTitle('');
+    setSelect(0);
+    setTouched(false);
+    setHasTitleError(false);
+    setHasSelectError(false);
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    if (hasTitleError && event.target.value) {
+      setHasTitleError(false);
+    }
+  };
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelect(+event.target.value);
+
+    if (hasSelectError && touched) {
+      setHasSelectError(false);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setHasTitleError(true);
+    }
+
+    if (select === 0) {
+      setHasSelectError(true);
+    }
+
+    if (title && select > 0) {
+      const user = [...users].filter(client => client.id === select);
+
+      const newPost = {
+        id: maxId + 1,
+        title: title,
+        completed: false,
+        userId: select,
+        user: user[0],
+      };
+
+      onAdd([...todos, newPost]);
+
+      reset();
+    }
+  };
```

Comment:
[REACT KNOWLEDGE] - It's great that you are using the `onSubmit` prop of the form to handle the form submission. This is a good practice because it allows the form to be submitted not only by clicking the submit button but also by pressing the enter key.